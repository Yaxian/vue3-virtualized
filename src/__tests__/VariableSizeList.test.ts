import { render } from '@testing-library/vue'

import VariableSizeList from '../VariableSizeList'

const Row = {
  name: 'Row',

  template: `
    <div>Row {{index}}</div>
  `,

  props: {
    index: {
      type: Number,
    },
    rowData: {

    },
  },

}

describe('VariableSizeList', () => {
  it('should render', async () => {
    const itemSize = jest.fn(() => 50)
    const TestComponent = {
      template: `
        <VariableSizeList
          :height="100"
          :item-size="itemSize"
          :item-count="200"
        >
          <template #default="{ data, index, key, isScrolling  }">
            <Row style="border: 1px solid" class="test-row" :index="index" :rowData="data" :isScrolling="isScrolling" />
          </template>
        </VariableSizeList>
      `,
      components: {
        VariableSizeList,
        Row,
      },
      setup() {
        return {
          itemSize,
        }
      },
    }

    const {
      container,
    } = render(TestComponent, {
      props: {

      },
    })

    expect(itemSize).toHaveBeenCalledTimes(4)
    expect(container).toMatchSnapshot()

    /**
     * jsdom doesn't support layout. This means measurements like this will always return 0 as it does here.
     * [issuecomment-481248489](https://github.com/testing-library/react-testing-library/issues/353#issuecomment-481248489)
     */

    // const outer = getByRole('vl-outer')
    // await fireEvent.scroll(outer, {
    //   target: {
    //     scrollTop: 100,
    //   },
    // })

    // await waitFor(() => new Promise(resolve => setTimeout(resolve, 2000)), { timeout: 3000 })
  })
})
