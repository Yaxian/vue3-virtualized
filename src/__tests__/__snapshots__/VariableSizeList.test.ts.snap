// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableSizeList should render 1`] = `
<div>
  <div
    role="vl-outer"
    style="position: relative; overflow: auto; will-change: transform; height: 100px; direction: ltr;"
  >
    <div
      style="height: 10000px; width: 100%;"
    >
      
      <div
        class="test-row"
        style="border: 1px solid; position: absolute; left: 0px; top: 0px; height: 50px; width: 100%;"
      >
        Row 0
      </div>
      
      
      <div
        class="test-row"
        style="border: 1px solid; position: absolute; left: 0px; top: 50px; height: 50px; width: 100%;"
      >
        Row 1
      </div>
      
      
      <div
        class="test-row"
        style="border: 1px solid; position: absolute; left: 0px; top: 100px; height: 50px; width: 100%;"
      >
        Row 2
      </div>
      
      
      <div
        class="test-row"
        style="border: 1px solid; position: absolute; left: 0px; top: 150px; height: 50px; width: 100%;"
      >
        Row 3
      </div>
      
    </div>
  </div>
</div>
`;
